eng/common/scripts/Package-Properties.ps1

```diff
@@ -154,17 +155,29 @@ function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgProps
 {
     foreach ($pkg in $activePkgList)
     {
+        LogDebug "Operating on $($pkg["name"])"
         $pkgProps = Get-PkgProperties -PackageName $pkg["name"] -ServiceDirectory $ServiceDirectory
-        $pkgPropsResult += $pkgProps
+        if ($null -ne  $pkgProps)
+        {
+            $pkgPropsResult += $pkgProps
+        }
     }
     return $pkgPropsResult
 }
 
 function Get-PkgListFromYml ($ciYmlPath)
 {
     $ProgressPreference = "SilentlyContinue"
-    Register-PSRepository -Default -ErrorAction:SilentlyContinue
-    Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+    if ((Get-PSRepository | ?{$_.Name -eq "PSGallery"}).Count -eq 0)
+    {
+        Register-PSRepository -Default -ErrorAction:SilentlyContinue
+    }
+
+    if ((Get-Module -ListAvailable -Name powershell-yaml | ?{$_.Version -eq "0.4.2"}).Count -eq 0)
+    {
+        Install-Module -Name powershell-yaml -RequiredVersion 0.4.2 -Force -Scope CurrentUser
```

Comment:
Doesn't powershell already check if the version is already installed and skip if it is? Or is the "-Force" flag preventing that?

Reply:
You are right, Am using the `-Force` flag to get past an "Untrusted Repository" prompt.